packages/rpc-core/src/rpc-methods/__tests__/get-signatures-for-address-test.ts

```diff
@@ -0,0 +1,57 @@
+import { createHttpTransport, createJsonRpc } from '@solana/rpc-transport';
+import type { SolanaJsonRpcErrorCode } from '@solana/rpc-transport/dist/types/json-rpc-errors';
+import type { Rpc } from '@solana/rpc-transport/dist/types/json-rpc-types';
+import fetchMock from 'jest-fetch-mock-fork';
+import { createSolanaRpcApi, SolanaRpcMethods } from '../../index';
+import { Base58EncodedAddress } from '@solana/keys';
+import { AllowedCommitmentForGetSignaturesForAddress } from '../getSignaturesForAddress';
+
+describe('getSignaturesForAddress', () => {
+    let rpc: Rpc<SolanaRpcMethods>;
+    beforeEach(() => {
+        fetchMock.resetMocks();
+        fetchMock.dontMock();
+        rpc = createJsonRpc<SolanaRpcMethods>({
+            api: createSolanaRpcApi(),
+            transport: createHttpTransport({ url: 'http://127.0.0.1:8899' }),
+        });
+    });
+
+    (['confirmed', 'finalized'] as AllowedCommitmentForGetSignaturesForAddress[]).forEach(commitment => {
+        describe(`when called with \`${commitment}\` commitment`, () => {
+            it('returns no transactions for a new address', async () => {
+                expect.assertions(1);
+                // This key is random, don't re-use in any tests that perform transactions
+                const publicKey = '3F6rba4VRgdGeYzgCNWQaEJUerUEQVVuwKrETigvHhJP' as Base58EncodedAddress;
+                const transactionsPromise = rpc.getSignaturesForAddress(publicKey, { commitment }).send();
+                await expect(transactionsPromise).resolves.toEqual([]);
+            });
+        });
+    });
+
+    describe('given an account with transactions', () => {
+        // TODO Need to be able to send transactions
+        it.todo('returns the transactions for that account');
+        it.todo('returns only the first transactions when called with a `limit`');
+        it.todo('returns transactions from `since` if defined');
+        it.todo('returns transactions only to `until` if defined');
```

Comment:
Thanks for leaving these here. I spent a few minutes digging around. What do you think the solution to this is?

* Make it so that you can start up the test validator with an array of transactions to play into it (eg. `--transaction-fixtures` or something)
* Literally play transactions into the validator when we start it up (eg. using cURL in `start-shared-test-validator.sh`)
* Create a one-time script that plays a bunch of transactions into the validator, then commit the `test-ledger/` directory that results.
* Something else?

Reply:
- I think the ideal here would be that first option, having the equivalent of `--account-dir` for transactions and us just having a directory of JSON-encoded transactions in this repo. I'm not sure what that'd look like in the validator though, and I'm not sure how we'd be able to deal with signatures. Feels like it needs a lot of design work and might need too much change in the `solana` codebase to be accepted though (thinking about the JSON -> transaction conversion mostly there, which I don't think currently exists in that codebase?)

- I'm a bit worried that doing it at the level of `sendTransaction` with curl or similar will be really tricky to debug, because we'll have just the encoded + signed transaction bytes being sent. Would be the same if we added fixtures to the validator at that level too. We'd be really reliant on documentation that nobody can easily review the correctness of, and if it was incorrect it could be really tricky to debug what's going on. Also would we be able to pre-generate transactions with valid blockhashes?

- I think I'd prefer to have transactions played on startup than commit the `test-ledger/` directory, just because it feels like a bit less indirection and would be easier to debug by just updating them and restarting. I'm guessing we won't end up needing enough transactions that we need to worry about the performance of playing them in CI, and I guess that'd be the main benefit of just checking in the ledger instead. 

- Would using a different Node transaction builder like `umi` for now be too circular? We'd have the benefit of being able to write our transactions in a Node script and easily modify/debug them, but we'd be relying on the correctness of that conversion to the transaction format.

Reply:
> I think I'd prefer to have transactions played on startup than commit the test-ledger/ directory…

I imagine that as the validator upgrades, the old `test-ledger/` directory will break and need to be regenerated?